docs/enhancements/short-url-enhancement.md

```diff
@@ -0,0 +1,118 @@
+---
+title: short-image-urls
+authors:
+  - "@mhrivnak"
+creation-date: 2022-11-30
+last-updated: 2022-12-06
+---
+
+# Short image URLs
+
+## Summary
+
+The assisted-image-service should be enhanced to serve discovery ISOs from URLs
+that are less than 255 characters and don't have query parameters. This will
+enable servers from multiple vendors to directly boot from an
+assisted-image-service URL using the baseboard management controller.
+
+## Motivation
+
+Servers from multiple vendors have limits to the length and composition of URLs
+from which their BMC can directly boot the system. But there is growing
+interest in the use case of directly booting such systems from a URL served by
+assisted-image-service, whether on-prem or from api.openshift.com.
+
+### Goals
+
+* Enable servers to boot discovery ISOs directly from a signed URL.
+
+### Non-Goals
+
+* 
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As the operator of bare metal servers, I can boot them from a signed URL served
+at api.openshift.com.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Requirements
+
+The latest vendor to request this has these requirements:
+* no URLs longer than 255 characters
+* no query parameters
+* no HTTP redirects
+* URL must end in ".iso"
+
+Those requirements appear similar to other vendors with limitations.
+
+#### signed URLs
+
+Today's signed URLs look like this:
+
+```
+https://api.openshift.com/api/assisted-images/images/739f38a1-5963-4511-8b5f-3c870a2c388c?arch=x86_64&image_token=<TOKEN>&type=minimal-iso&version=4.11
+```
+
+That should change to this:
+
+```
+https://api.openshift.com/api/assisted-images/bytoken/<TOKEN>/4.11/x86_64/minimal.iso
+```
+
+Because the infra-env ID is embedded in the token (which is a JWT), the
+infra-env ID can be removed from the URL path.
+
+#### unsigned URLs
+
+Today's unsigned URLs look like this:
+
+```
+http://api.openshift.com/api/assisted-images/images/881dd303-940e-4c1a-a844-dc3c64da0e95?arch=x86_64&type=full-iso&version=4.11
+```
+
+That should change to:
+
+```
+http://api.openshift.com/api/assisted-images/byid/881dd303-940e-4c1a-a844-dc3c64da0e95/4.11/x86_64/full.iso
+```
+
+#### Compatibility
+
+The previous URL layouts should continue to work. But the default URLs returned
```

Comment:
Do we keep the older format for backward compatibility? 
If yes, can we somehow mark (or just document) that the old format is deprecated?

Reply:
I don't think we can mark it as deprecated, but at the same time no one should be relying on the format.

The URL is always given to the user in full so they shouldn't care about how it's formatted. I see it as the same as when we moved from s3 to the image service, the URL will be different, but it should be transparent to the user.

Reply:
I think we need compatibility just long enough to make sure that any URL handed out recently continue to work after the service gets upgraded.